pkgs/development/python-modules/zerobin/default.nix

```diff
@@ -1,24 +1,28 @@
 { stdenv
 , buildPythonPackage
 , fetchFromGitHub
-, cherrypy
+, appdirs
+, beaker
+, bleach
 , bottle
-, lockfile
+, cherrypy
 , clize
+, lockfile
+, paste
 }:
 
 buildPythonPackage {
   pname = "zerobin";
-  version = "20160108";
+  version = "1.0.5";
 
   src = fetchFromGitHub {
     owner = "sametmax";
     repo = "0bin";
-    rev = "7da1615";
-    sha256 = "1pzcwy454kn5216pvwjqzz311s6jbh7viw9s6kw4xps6f5h44bid";
+    rev = "v1.0.5";
+    sha256 = "sha256-JvgPWkycHJApHZJcvf3Oxm0fq625jr0FJ5/6IRAc3rU";
   };
 
-  propagatedBuildInputs = [ cherrypy bottle lockfile clize ];
+  propagatedBuildInputs = [ appdirs beaker bleach bottle cherrypy clize lockfile paste ];
 
   # zerobin doesn't have any tests, but includes a copy of cherrypy which
```

Comment:
As put in the comment, I had to leave `doCheck = false`. but maybe you know a better way to avoid the error?

Reply:
I found [one previous case](
https://github.com/NixOS/nixpkgs/pull/44571#issuecomment-410944462) where `doCheck = false` was actually encouraged if there are no tests to run, so I think we're good with leaving it in.

Regarding the error:

> nix_run_setup runserver: Received extra arguments: test

It looks like the root cause of this is the same as what's described in this SO answer:   https://stackoverflow.com/a/52550011

- `buildPythonApplication`'s `checkPhase` runs `setup.py test`
- `setup.py test` tries to find tests by importing modules
- [`zerobin/__main__.py` runs the main function](https://github.com/Tygs/0bin/blob/v1.0.5/zerobin/__main__.py#L5) upon being imported
- Because `test` is not a recognized command, [`runserver` is assumed to be the desired command](https://github.com/Tygs/0bin/blob/v1.0.5/zerobin/cli.py#L218-L219), but it still fails because the `runserver` function isn't expecting an argument like `test`

Changing `__main__.py` to wrap the `main()` call with a guard like described in the SO answer should do away with the error.

```python
if __name__ == "__main__":
    main()
```

This can be a PR against upstream - or a patch in this package, but if we're leaving `doCheck = false` in, then it may not make much sense to maintain a patch for something that's not strictly required.

Reply:
@ento, ok I'm leaving `doCheck = false`, thanks!